service/controller/resource/verticalpodautoscaler/resource.go

```diff
@@ -135,6 +140,39 @@ func (r *Resource) getObject(ctx context.Context, v interface{}) (*vpa_types.Ver
 	return vpa, nil
 }
 
+func (r *Resource) getMaxCPU(ctx context.Context) (*resource.Quantity, error) {
+	nodes, err := r.k8sClient.K8sClient().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var nodeCpu *resource.Quantity
+	if len(nodes.Items) > 0 {
+		n := nodes.Items[0]
+		s, ok := n.Status.Allocatable[v1.ResourceCPU]
+		if ok {
+			nodeCpu = &s
+		}
+
+		for _, n := range nodes.Items[1:] {
+			s, ok := n.Status.Allocatable[v1.ResourceCPU]
+			if ok && nodeCpu.Cmp(s) == -1 {
+				nodeCpu = &s
+			}
+		}
+	}
+	if nodeCpu == nil || nodeCpu.IsZero() {
```

Comment:
If `len(nodes.Items) == 0`, then `nodeCpu` could be nil, which would panic is we just called `IsZero` on. Fixed below as well.

Reply:
Nice